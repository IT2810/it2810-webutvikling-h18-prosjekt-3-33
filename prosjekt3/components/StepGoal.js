import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';

export default class StepGoal extends React.Component {
  state = {
    stepGoal: 0,
  }
  componentDidMount(){
    this.setState({stepGoal: this.props.stepGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.stepGoal != prevProps.stepGoal){
      this.setState({stepGoal: this.props.stepGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of steps {this.props.stepGoal} </Text>
          <PedometerCounter stepGoal={this.state.stepGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.unfinishedGoalColor,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
