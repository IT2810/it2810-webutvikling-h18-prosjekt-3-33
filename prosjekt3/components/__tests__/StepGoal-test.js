import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MobileButton from 'react-native';
import StepGoal from '../StepGoal';
import { Pedometer } from 'expo';


describe('StepGoal', () => {

	it('renders correctly', () => {
	  	const stepgoal = renderer.create(<StepGoal />).toJSON();
	  	expect(stepgoal).toMatchSnapshot();
	});

});
