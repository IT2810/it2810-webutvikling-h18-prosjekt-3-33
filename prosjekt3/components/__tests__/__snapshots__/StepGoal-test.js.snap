// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepGoal renders correctly 1`] = `null`;
